integration/gateway/gateway_test.go

```diff
@@ -129,175 +187,104 @@ var _ = Describe("GatewayService", func() {
+		It("should respond with block number", func() {
+			_, transactionID := submitTransaction("respond", []byte("200"), []byte("conga message"), []byte("conga payload"))
+			firstStatus, err := commitStatus(transactionID, signingIdentity.Serialize, signingIdentity.Sign)
 			Expect(err).NotTo(HaveOccurred())
 
-			identity, err = signingIdentity.Serialize()
+			_, transactionID = submitTransaction("respond", []byte("200"), []byte("conga message"), []byte("conga payload"))
+			nextStatus, err := commitStatus(transactionID, signingIdentity.Serialize, signingIdentity.Sign)
 			Expect(err).NotTo(HaveOccurred())
-		})
 
-		AfterEach(func() {
-			conn.Close()
-			cancel()
+			Expect(nextStatus.BlockNumber).To(Equal(firstStatus.BlockNumber + 1))
 		})
 
-		It("should respond with status of submitted transaction", func() {
-			statusRequest := &gateway.CommitStatusRequest{
-				ChannelId:     "testchannel",
-				Identity:      identity,
-				TransactionId: transactionID,
-			}
-
-			statusRequestBytes, err := proto.Marshal(statusRequest)
-			Expect(err).NotTo(HaveOccurred())
+		It("should fail on unauthorized identity", func() {
+			_, transactionID := submitTransaction("respond", []byte("200"), []byte("conga message"), []byte("conga payload"))
+			badIdentity := network.OrdererUserSigner(network.Orderer("orderer"), "Admin")
+			_, err := commitStatus(transactionID, badIdentity.Serialize, signingIdentity.Sign)
+			Expect(err).To(HaveOccurred())
 
-			signature, err := signingIdentity.Sign(statusRequestBytes)
-			Expect(err).NotTo(HaveOccurred())
+			grpcErr, _ := status.FromError(err)
+			Expect(grpcErr.Code()).To(Equal(codes.PermissionDenied))
+		})
 
-			signedStatusRequest := &gateway.SignedCommitStatusRequest{
-				Request:   statusRequestBytes,
-				Signature: signature,
+		It("should fail on bad signature", func() {
+			_, transactionID := submitTransaction("respond", []byte("200"), []byte("conga message"), []byte("conga payload"))
+			badSign := func(digest []byte) ([]byte, error) {
+				return signingIdentity.Sign([]byte("WRONG"))
 			}
+			_, err := commitStatus(transactionID, signingIdentity.Serialize, badSign)
+			Expect(err).To(HaveOccurred())
 
-			actualStatus, err := gatewayClient.CommitStatus(ctx, signedStatusRequest)
-			Expect(err).NotTo(HaveOccurred())
-
-			expectedStatus := &gateway.CommitStatusResponse{
-				Result: peer.TxValidationCode_VALID,
-			}
-			Expect(proto.Equal(actualStatus, expectedStatus)).To(BeTrue(), "Expected\n\t%#v\nto proto.Equal\n\t%#v", actualStatus, expectedStatus)
+			grpcErr, _ := status.FromError(err)
+			Expect(grpcErr.Code()).To(Equal(codes.PermissionDenied))
 		})
+	})
 
-		It("should fail on unauthorized identity", func() {
-			badSigningIdentity := network.OrdererUserSigner(network.Orderer("orderer"), "Admin")
-			badIdentity, err := badSigningIdentity.Serialize()
+	Describe("ChaincodeEvents", func() {
+		It("should respond with emitted chaincode events", func() {
```

Comment:
I guess that you still plan to add security related tests, like the one in `CommitStatus` api ? However, that can always be added later.